kmipserver/server.go

```diff
@@ -24,19 +24,49 @@ type RequestHandler interface {
 	HandleRequest(ctx context.Context, req *kmip.RequestMessage) *kmip.ResponseMessage
 }
 
+// ConnectHook is a function that can be used to perform actions when a new connection is established.
+// It takes a context.Context as input and returns a modified context.Context.
+type ConnectHook func(context.Context) context.Context
+
+// TerminateHook is a function that can be used to perform cleanup actions when a connection is terminated.
+// It takes a context.Context as input.
+type TerminateHook func(context.Context)
```

Comment:
There's a caveat here that implementer must be aware of:

the context passed to the terminateHook my have been canceled. This means that it cannot be passed as is to context cancellable functions (http calls, database queries , etc...) without wrapping it into [context.WithoutCancel](https://pkg.go.dev/context#WithoutCancel) to silence the cancellation

Reply:
or maybe we must simply document it, but it would be more error prone